src/groupeddataframe/grouping.jl

```diff
@@ -366,9 +378,33 @@ _nrow(x::NamedTuple{<:Any, <:Tuple{Vararg{AbstractVector}}}) =
 _ncol(df::AbstractDataFrame) = ncol(df)
 _ncol(x::Union{NamedTuple, DataFrameRow}) = length(x)
 
-function gen_fun(::Type{NT}, f2) where NT
-    get_val(::Val{T}) where {T} = T
+struct Reduce{F, A}
+    f::F
+    adjust::A
+end
+Reduce(f) = Reduce(f, nothing)
+function (r::Reduce)(x)
+    v = reduce(r.f, x)
+    r.adjust === nothing ? v : r.adjust(v, length(x))
+end
+
+check_reduction(f::Any) = f
+check_reduction(::typeof(sum)) = Reduce(Base.add_sum)
+check_reduction(::typeof(prod)) = Reduce(Base.mul_prod)
+check_reduction(::typeof(maximum)) = Reduce(max)
+check_reduction(::typeof(minimum)) = Reduce(min)
+check_reduction(::typeof(mean)) = Reduce(Base.add_sum, /)
```

Comment:
If you wanted to follow my recommendation above here we would write:
```
check_reduction(::typeof(mean)) = Reduce(Base.add_sum, (s,x) -> s / length(x))
```

Reply:
See reply above.